app/customer.py

```diff
@@ -0,0 +1,74 @@
+from decimal import Decimal, ROUND_HALF_UP
+
+from app.car import Car
+
+
+class Customer:
+    def __init__(
+        self,
+        name: str,
+        product_cart: dict,
+        location: list,
+        money: Decimal,
+        car_characteristics: dict
+    ) -> None:
+        self.name = name
+        self.product_cart = product_cart
+        self.location = location
+        self.money = Decimal(str(money))
+        self.car = Car(**car_characteristics)
+        self.home_location = location
+
+    def find_cheapest(
+            self,
+            shops: list,
+            fuel_price: Decimal
+    ) -> tuple:
+        all_shops_total = {}
+        for shop in shops:
+            fuel_cost = self.count_fuel_cost(
+                shop.location,
+                self.car,
+                fuel_price
+            )
+            products_cost = self.count_prod_cost(shop.products)
+            total = ((fuel_cost + products_cost).
+                     quantize(Decimal("0.01"), rounding=ROUND_HALF_UP))
+
+            all_shops_total[shop] = total
+
+            print(f"{self.name}'s trip to the {shop.name} costs {total}")
+
+        cheapest_shop = min(all_shops_total, key=all_shops_total.get)
+        return cheapest_shop, all_shops_total[cheapest_shop]
+
+    def count_fuel_cost(
```

Comment:
this would be rather a Car method - not a customer method. it doesn't depend on a customer in any way so it would be logical to move it out. Your customer already has a lot of methods and having one class that does it all is not always a good idea